FinSearchUnified/Tests/ShopwareProcessTest.php

```diff
@@ -0,0 +1,110 @@
+class ShopwareProcessTest extends TestCase
+{
+    /**
+     * @throws \ReflectionException
+     * @throws \Exception
+     */
+    public function testProductStreams()
+    {
+        $categoryA = new Category();
+        $categoryA->setId(1);
+        $categoryA->setActive(true);
+
+        $categoryB = new Category();
+        $categoryB->setId(2);
+        $categoryB->setActive(true);
+        $categoryB->setStream(new ProductStream());
+
+        $categoryC = new Category();
+        $categoryC->setId(3);
+        $categoryC->setActive(false);
+        $categoryC->setStream(new ProductStream());
+        $categoryC->setParent($categoryB);
+
+        $categoryD = new Category();
+        $categoryD->setId(4);
+        $categoryD->setActive(true);
+        $categoryD->setStream(new ProductStream());
+        $categoryD->setParent($categoryB);
+
+        $childCategories = new ArrayCollection(
+            [$categoryC, $categoryD]
+        );
+
+        $children = new PersistentCollection(
+            Shopware()->Models(),
+            Shopware()->Models()->getClassMetadata(Category::class),
+            $childCategories
+        );
+
+        $categoryB->setChildren($children);
+
+        $categories = new ArrayCollection(
+            [$categoryA, $categoryB]
+        );
+
+        $parameters = new PersistentCollection(
+            Shopware()->Models(),
+            Shopware()->Models()->getClassMetadata(Category::class),
+            $categories
+        );
+
+        $mockRepository = $this->createMock(Repository::class);
+        $mockRepository->method('prepareCriteria');
+
+        $result = new \stdClass();
+        $result->totalCount = 10;
+        $result->products = [];
+
+        for ($i = 0; $i < 10; $i++) {
+            $product = new BaseProduct(rand(), rand(), uniqid());
+            $result->products[] = $product;
+        }
+
+        $mockSearchResult = $this->createMock(ProductNumberSearchResult::class);
+        $mockSearchResult->method('getProducts')->willReturn($result->products);
+
+        $contextService = Shopware()->Container()->get('shopware_storefront.context_service');
+
+        $mockProductNumberSearch = $this->createMock(ProductNumberSearch::class);
+        $mockProductNumberSearch->expects($this->atLeastOnce())->method('search')->willReturn($mockSearchResult);
+
+        /** @var Repository $mockRepository */
+        $shopwareProcess = new ShopwareProcess(
+            Shopware()->Container()->get('cache'),
+            $mockRepository,
+            $contextService,
+            $mockProductNumberSearch
+        );
+
+        $reflector = new ReflectionObject($shopwareProcess);
+        $method = $reflector->getMethod('parseProductStreams');
+        $method->setAccessible(true);
+        $articles = $method->invoke($shopwareProcess, $parameters);
+
+        foreach ($articles as $article) {
+            foreach ($article as $category) {
+                $this->assertThat($category, $this->logicalOr(
```

Comment:
* Do not only check if the categories are present but also if the correct categories are assigned to the product (by checking the array index which should be the product's ID).
* Theoretically the `logicalOr` shouldn't be necessary since `$categoryB` and `$categoryD` should always be present.

So for the assertions:
* Check if the keys of `$articles` contain only the IDs of the created base products.
* Check if each entry of `$articles` has both `$categoryB` and `$categoryD`.